.github/workflows/scripts/e2e.container.default.verify.sh

```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/env bash
+
+# shellcheck source=/dev/null
+source "./.github/workflows/scripts/e2e-verify.common.sh"
+
+go env -w GOFLAGS=-mod=mod
+
+# verify_provenance_content verifies provenance content generated by the container generator.
+verify_provenance_content() {
+    ATTESTATION=$(jq -r '.payload' <"$PROVENANCE" | base64 -d)
+
+    echo "  **** Provenance content verification *****"
+
+    # Verify all common provenance fields.
+    e2e_verify_common_all "$ATTESTATION"
+
+    e2e_verify_predicate_subject_name "$ATTESTATION" "$CONTAINER"
+    e2e_verify_predicate_builder_id "$ATTESTATION" "https://github.com/slsa-framework/slsa-github-generator/.github/workflows/generator_container_slsa3.yml@refs/heads/main"
+    e2e_verify_predicate_buildType "$ATTESTATION" "https://github.com/slsa-framework/slsa-github-generator/generic@v1"
+}
+
+THIS_FILE=$(e2e_this_file)
+BRANCH=$(echo "$THIS_FILE" | cut -d '.' -f4)
+echo "branch is $BRANCH"
+echo "GITHUB_REF_NAME: $GITHUB_REF_NAME"
+echo "GITHUB_REF_TYPE: $GITHUB_REF_TYPE"
+echo "GITHUB_REF: $GITHUB_REF"
+echo "DEBUG: file is $THIS_FILE"
+
+# Verify provenance authenticity.
+e2e_run_verifier_all_releases "HEAD"
+
+# TODO: Add provenance content verification for containers with 
+# with provenance attached on the OCI registry.
+if [[ -n $PROVENANCE ]]; then
+    verify_provenance_content
+fi
```

Comment:
nit: For container e2e `$PROVENANCE` is empty is it not? Maybe I was misunderstanding something? If so this looks like dead code.

Reply:
With your suggestion to check based on e2e workflow name for what type of build it was, we can now check provenance content correctly. I extract it in the workflow with cosign, and it will not be set in arguments in the CLI verify script.

Reply:
I think you resolved this? feel free to unresolve if not.